### DISCUSSION TITLE
这个是否需要uiautomator2类似的apk的

### DISCUSSION THREAD

**Discussion by aogg** [SuperMonster003/AutoJs6#258] [General]
还是纯adb的

**Comment by aogg** [comment]
如果不将脚本打包为apk能用吗？是不是打包为apk才必须要无障碍服务的

**Comment by SuperMonster003** [comment]
对于标题及评论中涉及的内容, 我无法明确理解具体的含义.

我尝试解答一下, 但可能出现文不对题的情况. 如有必要, 可重新表述上述全部或部分问题.

AutoJs6 可以实现 UiAutomator2 类似的功能, 它是基于安卓无障碍服务实现的. 除此之外, 也可以使用 Shell 指令借助 Root 权限 (或直接使用 Shizuku 提升权限) 来发送事件实现点击/滑动/滚动/文本输入等操作.

AutoJs6 主要功能就是作为一个 JavaScript IDE 实现一些功能 (详见 README.md 或应用文档), 因此不依赖于打包功能. 打包功能可以将脚本, 包括一些资源整合到一起, 安装到设备中, 这样可以独立于 AutoJs6 成为一个标准应用存在并执行指定代码.

无障碍服务可以由 AutoJs6 全局控制, 当关闭时, 所有脚本均无法使用无障碍服务 (当然也可以在脚本中使用代码控制无障碍服务). 打包应用有独立的无障碍物服控制开关, 它只针对当前打包应用, 对 AutoJs6 中执行的脚本没有影响.
